### PR TITLE
DOC: Removed Notes from DataFrame.applymap

### LINKED ISSUE #31620: Does #28827 also apply to applymap?
In  #28854, the note 

> In the current implementation applymap calls func twice on the first column/row to decide whether it can take a fast or slow code path. This can lead to unexpected behavior if func has side-effects, as they will take effect twice for the first column/row.

was removed from the `DataFrame.apply` docstring.

However, it is still in the [docstring](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.applymap.html?highlight=applymap#pandas.DataFrame.applymap) of `DataFrame.applymap`.

Does it need to be removed there too?

EDIT
----

As far as I can tell, this behaviour (calling the function twice to decide which path to take) still happens for `groupby.apply` and is what a user was asking about in #31111. Does it need to be documented in Groupby.Apply? Do end-users need to know about such internals?

### PR DESCRIPTION
**Pull request by r0cketr1kky**
- closes #31620 

Documentation screenshot: 
Screenshot from 2020-02-05 17-08-33

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
Thanks for the PR! There's a linting issue at https://github.com/pandas-dev/pandas/pull/31695/checks?check_run_id=427378996#step:10:25.

I think you may need o remove a newline somewhere. Not entirely sure where though.

**Comment by r0cketr1kky** [comment]
Please review this @TomAugspurger

**Comment by MarcoGorelli** [comment]
Thanks @r0cketr1kky 

Looks like you've checked in some unwanted files (e.g.  `doc/make.spec`) - could you please [unstage them](https://docs.gitlab.com/ee/university/training/topics/unstage.html)?

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6990,8 +6982,7 @@ def applymap(self, func) -> "DataFrame":
         0  3  4
         1  5  5
 
-        Note that a vectorized version of `func` often exists, which will
-        be much faster. You could square each number elementwise.
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: This part doesn't need changing, the vectorised version is still faster:
```
In [24]: s = pd.Series(np.random.randn(10000)).apply(lambda x: x**2)                                                                                         

In [25]: %timeit s.apply(lambda x: x**2)                                                                                                                     
6.33 ms ± 253 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [26]: %timeit s**2                                                                                                                                        
238 µs ± 803 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [27]: pd.testing.assert_series_equal(s.apply(lambda x: x**2), s**2)
```

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6990,8 +6982,7 @@ def applymap(self, func) -> "DataFrame":
...
+        You could square each number elementwise.
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: This part doesn't need changing, the vectorised version is still faster:
```
In [28]: s = pd.Series(np.random.randn(10000))                                                                                                               

In [29]: %timeit s.apply(lambda x: x**2)                                                                                                                     
6.39 ms ± 5.73 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [30]: %timeit s**2                                                                                                                                        
243 µs ± 4.38 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [31]: pd.testing.assert_series_equal(s.apply(lambda x: x**2), s**2) 
```

Comment (r0cketr1kky) [reply pandas/core/frame.py]: Oh okay. Thanks a lot. Included it.

**Comment by r0cketr1kky** [comment]
Please review this!

File: pandas/core/generic.py [hunk 3]
```diff
@@ -10329,6 +10329,26 @@ def _doc_parms(cls):
 %(examples)s
 """
 
+_num_doc_mad = """
+%(desc)s
```
Comment (jreback) [inline pandas/core/generic.py]: this is an unrelated change, can you revert

**Review by jreback — CHANGES_REQUESTED** [review]
if you can only have the 1 change in this PR

File: pandas/core/frame.py [hunk 4]
```diff
@@ -6969,14 +6969,6 @@ def applymap(self, func) -> "DataFrame":
         --------
         DataFrame.apply : Apply a function along input axis of DataFrame.
 
-        Notes
-        -----
-        In the current implementation applymap calls `func` twice on the
```
Comment (jbrockmendel) [inline pandas/core/frame.py]: is this no longer accurate?  we regularly get questions about the user-defined functions being called in groupby.apply/agg to determine fast vs slow path

Comment (MarcoGorelli) [reply pandas/core/frame.py]: In #28854 this was removed from `DataFrame.apply`, and there are comments on the original issue (#28827) saying that the implementation has changed recently, and `DataFrame.applymap` directly calls `DataFrame.apply`

**Comment by datapythonista** [comment]
This seems stale. @r0cketr1kky if you want to finish this (mainly remove the unrelated changes), please open a PR or ping us here to reopen. Thanks!

**Comment by r0cketr1kky** [comment]
Sorry for being late! I just noticed this @datapythonista ! Can you reopen this? I'll revert back with the changes.

**Comment by jreback** [comment]
@r0cketr1kky can you merge master and update

File: pandas/core/frame.py [hunk 5]
```diff
@@ -6990,8 +6982,9 @@ def applymap(self, func) -> "DataFrame":
         0  3  4
         1  5  5
 
-        Note that a vectorized version of `func` often exists, which will
-        be much faster. You could square each number elementwise.
+        Note that a vectorized version of `func` often exists,
+        which will be much faster.
+        You could square each number elementwise.
```
Comment (jbrockmendel) [inline pandas/core/frame.py:6987]: can you revert this

Comment (r0cketr1kky) [reply pandas/core/frame.py:6987]: Sorry, for being late! I'm having a very busy semester. I'll do this asap and get back!

**Comment by alonme** [comment]
It doesn't look to me like this issue is actually solved.
I hope to get it solved in #34183

**Comment by MarcoGorelli** [comment]
Closing as #34183 was merged and closed the issue - congrats @alonme for a really impressive contribution!
